man/po/da.po

```diff
@@ -20,7 +20,7 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: shadow-man\n"
-"POT-Creation-Date: 2016-09-18 14:03-0500\n"
+"POT-Creation-Date: 2024-03-14 18:23-0500\n"
```

Comment:
Are these things purely generated, or do they need some manual intervention?

If they are generated, why are they under git control, and not just generated by the build system and gitignored?

Reply:
They are generated, but they take a long time to build.

Reply:
How much is long?  And how can I build them myself?

Reply:
(It would be nice to mention that in the commit message, as in:)

```
Update translations

Scripted change:

$ foo

Signed-off-by: ...
```

Reply:
The POT-creation-date is automatically added by itstool.

Reply:
I don't mean just the creation date.  I mean how can I regenerate the POT files, to do this update in the future, if I need to.  Ans also to see how much time it takes in my computer.

Reply:
For example, in the Linux man-pages, I do have a generated page, from kernel sources, and I document how it's updated in the commit message:

```
commit df9f1bc4cafc9f871c320df76c24b01ba930cafd
Author: Alejandro Colomar <alx@kernel.org>
Date:   Sat Mar 16 22:57:05 2024 +0100

    bpf-helpers.7: Refresh page
    
    Scripted change:
    
    $ ~/src/linux/linux/6.8/scripts/bpf_doc.py | rst2man > man7/bpf-helpers.7
    
    Signed-off-by: Alejandro Colomar <alx@kernel.org>

```

Reply:
See the first line of man/po/shadow-man-pages.pot :

```
    1 # To re-generate, run "cd man/po; make update-po"
```

It gets auto-inserted by the Makefile.

I'm not disagreeing with the goal of better documentation.  But this PR is just about making sure that 'dist clean' works and that make distcheck passes.  It'd be great if we found someone experienced in itstool and msgmerge and all the translations stuff to maintain that.

Reply:
Ahhh, thanks!

I was being picky about this because a patch set that fixes distcheck somehow manages to change 100k+ lines.  :)

But yeah, I guess that's fine for now.

Reply:
The problem is that translations have two sources of input:
- they scrape contents out of your source code (`*.c`) and duplicate them into the .po files (duplicated across ALL the files, in fact)
- they contain the actual unique translated message strings provided by the translator team

You actually don't need to regularly sync the first type of contents, but you do need the second type or else you simply have no translations. Also, they do need to be in git or else people who build from a git clone don't have translated software.

That being said, it's not typically considered ideal to regularly run the update-po target. Instead, it should be run once, shortly before release, after declaring a string freeze. At that point you update the .po sources, send them out to translators, and then merge them back in right before release.

Reply:
Yup, right now we just get occasional (highly appreciated) contributions from one translator at a time, and commit them as they come in.  We could try, next time, generating the .po sources at the first 4.16.0-rc0 prerelease, with the hopes of getting translations before final 4.16.